Tests/RxCocoaTests/SharedSequence+OperatorTest.swift

```diff
@@ -217,6 +217,7 @@ extension SharedSequenceOperatorTests {
         let hotObservable = BackgroundThreadPrimitiveHotObservable<Int>()
 
         var events = [Event<Int>]()
+        var afterNextEvents = [Int]()
```

Comment:
This doesn't test the ordering properly.

Please turn this into `var events = [OrderedEvent]()`

````
struct OrderedEvent {
    case before(Event<Int>)
    case subscription(Event<Int>)
    case after(Event<Int>)
}

hotObservable.asDriver(onErrorJustReturn: -1).do(onAfterNext: { e in
            XCTAssertTrue(DispatchQueue.isMain)
            events.append(.after(.next(e)))
        })
        .do(onNext: { e in
                 events.push(.subscribe(.next(e)))
        }
```

in the end please assert that events are equal to some order.